tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -1120,11 +1129,12 @@ void TangoRosNode::PublishMesh() {
         }
       }
       // Publish Tango mesh as occupancy grid.
-      if (occupancy_grid_publisher_.getNumSubscribers() > 0) {
+      if (enable_3dr_occupancy_grid_) {
         occupancy_grid_.data.clear();
         if (tango_3d_reconstruction_helper::ExtractFloorPlanImageAndConvertToOccupancyGrid(
             t3dr_context_, time_offset_, start_of_service_frame_id_,
-            t3dr_resolution_, t3dr_occupancy_grid_threshold_, &occupancy_grid_))
+            t3dr_resolution_, t3dr_occupancy_grid_threshold_, &occupancy_grid_) &&
+            occupancy_grid_publisher_.getNumSubscribers() > 0)
```

Comment:
Yes we do. It can happen that there is no subscriber for the occupancy grid but we still want to extract it to be able to save it.